src/components/app/app.js

```diff
@@ -38,8 +39,8 @@ class App extends React.Component {
     if (prevProps.data !== this.props.data) {
```

Comment:
How is the app getting updated when data changes ? Even data seems to be an object like options. Here we are again comparing only references

Reply:
@ravi-kumar-pilla This works mainly because every time the data is modified, it is changed into a new object. Therefore, when comparisons are made, they are between new references.